numpy/testing/_private/utils.py

```diff
@@ -388,6 +388,12 @@ def assert_equal(actual, desired, err_msg='', verbose=True):
 
     except (TypeError, ValueError, NotImplementedError):
         pass
+    except DeprecationWarning:
+        # version 1.18
+        # gisnan used to raise for datetime64/timedelta64 (dt64) since they had
+        # no isnan implementation. That call no longer raises. However, a
+        # DeprecationWarning is raised when comparing dt64 to 0.
+        pass
 
     try:
         isdesnat = isnat(desired)
```

Comment:
The assertion error on line 402/408 below is now unreachable - `assert_equal(datetime64('nat'), timedelta64('nat'))` used to assert and no longer does.

Reply:
The logic on lines 379-382 now fails, since `gisnan` returns True for Nat as well as Nan. Adding a test and rearranging the logic to test for NaT before Nan.